File: examples/SignatureRequestBulkCreateEmbeddedWithTemplateExample.rb
require "dropbox-sign"

Dropbox::Sign.configure do |config|
  # Configure HTTP basic authorization: api_key
  config.username = "YOUR_API_KEY"

  # or, configure Bearer (JWT) authorization: oauth2
  # config.access_token = "YOUR_ACCESS_TOKEN"
end

signature_request_api = Dropbox::Sign::SignatureRequestApi.new

signer_list_1_signer = Dropbox::Sign::SubSignatureRequestTemplateSigner.new
signer_list_1_signer.role = "Client"
signer_list_1_signer.name = "George"
signer_list_1_signer.email_address = "george@example.com"
signer_list_1_signer.pin = "d79a3td"

signer_list_1_custom_fields = Dropbox::Sign::SubBulkSignerListCustomField.new
signer_list_1_custom_fields.name = "company"
signer_list_1_custom_fields.value = "ABC Corp"

signer_list_1 = Dropbox::Sign::SubBulkSignerList.new
signer_list_1.signers = [signer_list_1_signer]
signer_list_1.custom_fields = [signer_list_1_custom_fields]

signer_list_2_signer = Dropbox::Sign::SubSignatureRequestTemplateSigner.new
signer_list_2_signer.role = "Client"
signer_list_2_signer.name = "Mary"
signer_list_2_signer.email_address = "mary@example.com"
signer_list_2_signer.pin = "gd9as5b"

signer_list_2_custom_fields = Dropbox::Sign::SubBulkSignerListCustomField.new
signer_list_2_custom_fields.name = "company"
signer_list_2_custom_fields.value = "123 LLC"

signer_list_2 = Dropbox::Sign::SubBulkSignerList.new
signer_list_2.signers = [signer_list_2_signer]
signer_list_2.custom_fields = [signer_list_2_custom_fields]

cc_1 = Dropbox::Sign::SubCC.new
cc_1.role = "Accounting"
cc_1.email_address = "accounting@example.com"

data = Dropbox::Sign::SignatureRequestBulkCreateEmbeddedWithTemplateRequest.new
data.client_id = "1a659d9ad95bccd307ecad78d72192f8"
data.template_ids = ["c26b8a16784a872da37ea946b9ddec7c1e11dff6"]
data.subject = "Purchase Order"
data.message = "Glad we could come to an agreement."
data.signer_list = [signer_list_1, signer_list_2]
data.ccs = [cc_1]
data.test_mode = true

begin
  result = signature_request_api.signature_request_bulk_create_embedded_with_template(data)
  p result
rescue Dropbox::Sign::ApiError => e
  puts "Exception when calling Dropbox Sign API: #{e}"
end

File: examples/SignatureRequestBulkCreateEmbeddedWithTemplateExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.*;
import com.dropbox.sign.auth.*;
import com.dropbox.sign.JSON;
import com.dropbox.sign.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class SignatureRequestBulkCreateEmbeddedWithTemplateExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");

        var signerList2CustomFields1 = new SubBulkSignerListCustomField();
        signerList2CustomFields1.name("company");
        signerList2CustomFields1.value("123 LLC");

        var signerList2CustomFields = new ArrayList<SubBulkSignerListCustomField>(List.of (
            signerList2CustomFields1
        ));

        var signerList2Signers1 = new SubSignatureRequestTemplateSigner();
        signerList2Signers1.role("Client");
        signerList2Signers1.name("Mary");
        signerList2Signers1.emailAddress("mary@example.com");
        signerList2Signers1.pin("gd9as5b");

        var signerList2Signers = new ArrayList<SubSignatureRequestTemplateSigner>(List.of (
            signerList2Signers1
        ));

        var signerList1CustomFields1 = new SubBulkSignerListCustomField();
        signerList1CustomFields1.name("company");
        signerList1CustomFields1.value("ABC Corp");

        var signerList1CustomFields = new ArrayList<SubBulkSignerListCustomField>(List.of (
            signerList1CustomFields1
        ));

        var signerList1Signers1 = new SubSignatureRequestTemplateSigner();
        signerList1Signers1.role("Client");
        signerList1Signers1.name("George");
        signerList1Signers1.emailAddress("george@example.com");
        signerList1Signers1.pin("d79a3td");

        var signerList1Signers = new ArrayList<SubSignatureRequestTemplateSigner>(List.of (
            signerList1Signers1
        ));

        var signerList1 = new SubBulkSignerList();
        signerList1.customFields(signerList1CustomFields);
        signerList1.signers(signerList1Signers);

        var signerList2 = new SubBulkSignerList();
        signerList2.customFields(signerList2CustomFields);
        signerList2.signers(signerList2Signers);

        var signerList = new ArrayList<SubBulkSignerList>(List.of (
            signerList1,
            signerList2
        ));

        var ccs1 = new SubCC();
        ccs1.role("Accounting");
        ccs1.emailAddress("accounting@example.com");

        var ccs = new ArrayList<SubCC>(List.of (
            ccs1
        ));

        var signatureRequestBulkCreateEmbeddedWithTemplateRequest = new SignatureRequestBulkCreateEmbeddedWithTemplateRequest();
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.clientId("1a659d9ad95bccd307ecad78d72192f8");
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.templateIds(List.of (
            "c26b8a16784a872da37ea946b9ddec7c1e11dff6"
        ));
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.message("Glad we could come to an agreement.");
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.subject("Purchase Order");
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.testMode(true);
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.signerList(signerList);
        signatureRequestBulkCreateEmbeddedWithTemplateRequest.ccs(ccs);

        try
        {
            var response = new SignatureRequestApi(config).signatureRequestBulkCreateEmbeddedWithTemplate(
                signatureRequestBulkCreateEmbeddedWithTemplateRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling SignatureRequestApi#signatureRequestBulkCreateEmbeddedWithTemplate");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}
